lib/reline/io/ansi.rb

```diff
@@ -245,39 +245,30 @@ def set_screen_size(rows, columns)
     self
   end
 
-  def cursor_pos
-    if both_tty?
-      res = +''
-      m = nil
-      @input.raw do |stdin|
-        @output << "\e[6n"
-        @output.flush
-        loop do
-          c = stdin.getc
-          next if c.nil?
-          res << c
-          m = res.match(/\e\[(?<row>\d+);(?<column>\d+)R/)
-          break if m
-        end
-        (m.pre_match + m.post_match).chars.reverse_each do |ch|
-          stdin.ungetc ch
+  private def cursor_pos_internal(timeout:)
+    match = nil
+    @input.raw do |stdin|
+      @output << "\e[6n"
+      @output.flush
+      timeout_at = Time.now + timeout
+      buf = +''
+      while (wait = timeout_at - Time.now) > 0 && stdin.wait_readable(wait)
+        buf << stdin.readpartial(1024)
```

Comment:
Why did this `stdin.readpartial(1024)` become necessary?

Reply:
When an io is ready for read, normally several bytes are readable. `readpartial` is more efficient than calling `wait_readable` and `getc` many times.

The code `buf = match.pre_match + match.post_match` assumes post_match exists. It assumes `res << input` part will append long bytes. I think the original intention of this method is to use readpartial instead of getc.

Reply:
Using `readpartial` is more efficient, I understand now. Thanks!